src/Streamly/Coreutils/Touch.hs

```diff
@@ -0,0 +1,48 @@
+-- |
+-- Module      : Streamly.Coreutils.Touch
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Update the access and modification times of a file to the current time.
+
+module Streamly.Coreutils.Touch
+    (
+      Touch
+    , touch
+    , setCreateNew
+    , setDeRef
+    )
+where
+
+import Streamly.Coreutils.Common (Switch(..))
+import Streamly.Coreutils.FileTest (test, isExisting)
+import qualified System.Posix.Files as Posix
+import Control.Monad (when, unless)
+
+data Touch = Touch
+    {
+      createNew :: Switch
+    , deRef :: Switch   -- touch the referenced file for symbolic link
+    }
+
+defaultConfig :: Touch
+defaultConfig = Touch On On
+
+setDeRef :: Switch -> Touch -> Touch
+setDeRef opt cfg = cfg {deRef = opt}
```

Comment:
Can we call it `followLinks` to keep it similar to other commands in this package?